StudentManagementSystemProject/src/main/java/com/zemoso/controller/StudentController.java

```diff
@@ -0,0 +1,122 @@
+package com.zemoso.controller;
+
+import java.util.List;
+
+import com.zemoso.dto.StudentDto;
+import com.zemoso.entity.Course;
+import com.zemoso.service.CourseService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.propertyeditors.StringTrimmerEditor;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.*;
+
+import com.zemoso.entity.Student;
+import com.zemoso.service.StudentService;
+
+import javax.validation.Valid;
+
+@Controller
+@RequestMapping("/students")
+public class StudentController {
+
+	private StudentService studentService;
+
+	@Autowired
+	private CourseService courseService;
+
+	public StudentController(StudentService theStudentService) {
+		studentService = theStudentService;
+	}
+
+
+	@InitBinder
+	public void initBinder(WebDataBinder dataBinder) {
+
+		StringTrimmerEditor stringTrimmerEditor = new StringTrimmerEditor(true);
+
+		dataBinder.registerCustomEditor(String.class, stringTrimmerEditor);
+	}
+
+	@GetMapping("/list")
+	public String listStudents(Model theModel) {
+
+		List<StudentDto> theStudents = studentService.findAll();
+
+		theModel.addAttribute("students", theStudents);
+		
+		return "students/list-students";
+	}
+	
+	@GetMapping("/showFormForAdd")
+	public String showFormForAdd(Model theModel) {
+
+		List<Course> courses=courseService.findAll();
+
+		Student theStudent = new Student();
```

Comment:
Can't we use Dto here? any specific reason to use Student class?

Reply:
Ya We can use it i missed it to replace with studentDto. I have changed it.